.gitattributes

```diff
@@ -0,0 +1,44 @@
+###############################
+# Git Line Endings            #
+###############################
+
+# Set default behaviour to automatically normalize line endings.
+* text=auto
```

Comment:
I looked over at the git documentation and found this line to be of questionable advice.

The `*` means apply to all.
The `text` part means treat `*` as text.

This translates to changing all things to text, including binary files, like for exampl a PNG or JPG file.

If there is anything that should be treated as a binary, this breaks that behavior.
This could then cause any binary file to be incorrectly altered.

I belive the Git Large File System comments below is a fix to the problem they caused with this line.

They have to remove the "text" interpretation that they forcibly added here.

Here is a quote:

> The attributes allow a fine-grained control, how the line endings are converted. Here is an example that will make Git normalize .txt, .vcproj and .sh files, ensure that .vcproj files have CRLF and .sh files have LF in the working directory, and prevent .jpg files from being normalized regardless of their content.
```
* text=auto
...
*.jpg -text
```

It might be safer to be explicit rather than wildcarding on all:
```
*.js text=auto
*.ts text=auto
*.java text=auto
*.txt text=auto
*.cmd text=auto
*.bat text=auto
...
```

References:
- https://www.git-scm.com/docs/gitattributes#_text
- https://www.git-scm.com/docs/gitattributes#_end_of_line_conversion

Reply:
Thanks for the review.  According to the article and source of this file, the LFS sections account for them to avoid the auto changing them. I have no strong preference and will likely reconsider this file in the future. For now, I think provides a good example and reference.

https://github.com/TAMULib/scholars-angular/pull/93/commits/43c295e4562779c430db3cc5371d5fb7458ede11 => https://rehansaeed.com/gitattributes-best-practices/

> As you saw above, we already do this with Git LFS but if you don't use LFS, read on as you may need to explicitly list binary files in certain rare circumstances.

https://www.git-scm.com/docs/gitattributes
https://en.wikipedia.org/wiki/Newline

Reply:
We may want to consider like you said be more explicit per file and more so grouping per project. Such as, https://github.com/alexkaratarakis/gitattributes

Reply:
https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

Reply:
This looks rather comprehensive, https://gist.github.com/nwlucas/333e30e0340f7a6b8594cc62da31f5a8.

Reply:
That "rather comprehensive one" still has:
```
* text=auto
```
Which makes all of the lines there explicitly adding `text` rather redundant.